src/v/compat/cluster_json.h

```diff
@@ -604,6 +604,13 @@ inline void rjson_serialize(
       w,
       "initial_retention_local_target_ms",
       tps.initial_retention_local_target_ms);
+    if (tps.mpx_virtual_cluster_id) {
+        write_member(
+          w,
+          "virtual_cluster_id",
```

Comment:
question: should this `mpx_virtual_cluster_id`?

Reply:
If you do the suggestion above, you can do `write_member(w, "mpx_virtual_cluster_id", tps.mpx_virtual_cluster_id)`